discord_signup.py

```diff
@@ -0,0 +1,204 @@
+# This is a test file for a user sign-up flow.
+# It requires the bot to be alive at all times, so I have not merged it into the main script (yet?)
+
+import configparser
+import os
+import re
+from typing import Callable, Optional
+import asyncio
+import discord
+from discord.ext import commands
+
+ONE_MINUTE = 60
+SIGNUP_TIMEOUT = ONE_MINUTE * 10
+
+# Read the settings.cfg file
+config_path = os.path.join(os.path.dirname(__file__), "settings.cfg")
+config = configparser.ConfigParser()
+config.read(config_path)
+
+# Add sections to config if they are missing
+if "DISCORD_ID_BY_EMAIL" not in config:
+    config["TODOIST_KEY_BY_EMAIL"] = {}
+
+if "DISCORD_ID_BY_EMAIL" not in config:
+    config["DISCORD_ID_BY_EMAIL"] = {}
+
+with open(config_path, "w") as configfile:
+    config.write(configfile)
+
+# Load the Discord bot token and channel ID from the config file
+try:
+    DISCORD_TOKEN = config.get("DISCORD", "TOKEN")
+    CHANNEL_ID = config.getint("DISCORD", "CHANNEL_ID")
+    SERVER_ID = config.getint("DISCORD", "SERVER_ID")
+    LAST_ONLINE = config.get("DISCORD", "LAST_ONLINE", fallback=None)
+except (configparser.NoSectionError, configparser.NoOptionError) as _:
+    print("Discord config set incorrectly")
+    exit()
+
+try:
+    TODOIST_LINK = config.get("TODOIST_AUTH", "APP_LINK")
+except (configparser.NoSectionError, configparser.NoOptionError) as _:
+    print("Todoist App link not set")
+    exit()
+
+
+intents = discord.Intents.default()
+intents.message_content = True
+intents.members = True
+bot = commands.Bot(intents=intents, command_prefix="!")
+
+EMAIL_REGEX = re.compile(r"[\w.+-]+@[\w-]+\.[\w.-]+")
+
+
+@bot.event
+async def on_ready():
+    print("Bot is ready")
+    synced = await bot.tree.sync()
+    print(synced)
+
+
+@bot.tree.command(name="signup")
+@discord.app_commands.describe(user_to_signup="Mention of user")
+async def signup(interaction: discord.Interaction, user_to_signup: discord.Member):
+    # The slash command gives a hint to the user about the mentions arg,
+    # but it isn't directly consumed in the function
```

Comment:
I'm not sure I understand what you mean by consumed in the function

Reply:
relic comment, will remove